### PR TITLE
More user friendly error message when PFS request fails

### PR DESCRIPTION
**Pull request by palango**
Fixes: #5064 

## Description

Displays a nicer error message instead of some cryptic parser error.

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Documentation
    - [ ] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [ ] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

File: raiden/network/pathfinding.py [hunk 1]
```diff
@@ -152,7 +152,8 @@ def get_pfs_info(url: str) -> PFSInfo:
             version=infos["version"],
         )
     except (json.JSONDecodeError, requests.exceptions.RequestException, KeyError) as e:
-        raise ServiceRequestFailed(str(e)) from e
+        msg = "Selected Pathfinding service returned unexpected reply"
```
Comment (Dominik1999) [inline raiden/network/pathfinding.py]: We seem to have three different notations of "pathfinding service". In the spec we call it "Pathfinding Service" in the Header in the text we note "path finding service" here it is "Pathfinding service". Can you make this consistent?

File: raiden/tests/unit/test_pfs_integration_refactored.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def test_get_pfs_info_error():
         with pytest.raises(ServiceRequestFailed) as error:
             get_pfs_info("url")
 
-        assert "Expecting property name enclosed in double quotes:" in str(error.value)
+        assert "Selected Pathfinding service returned unexpected reply" in str(error.value)
```
Comment (lullis) [inline raiden/tests/unit/test_pfs_integration_refactored.py]: Is the indentation level correct? Shouldn't this assert be part of the second, inner `with` block?

Comment (lullis) [inline raiden/tests/unit/test_pfs_integration_refactored.py]: If the string is a constant, why the `in` comparison and not a simpler `==`?

Comment (palango) [reply raiden/tests/unit/test_pfs_integration_refactored.py]: I don't think so

File: raiden/network/pathfinding.py (hunk 1 above)
Comment (palango) [reply raiden/network/pathfinding.py]: Done

**Comment by palango** [comment]
@Dominik1999 you looked at an old version, in the up-to-date script checkhttps://github.com/raiden-network/raiden/pull/5126/files#diff-cef6bfb366cd652b5a560e1b252629f4R155

**Pull request merged**